src/test/logical_query_plan/lqp_translator_test.cpp

```diff
@@ -399,6 +400,57 @@ TEST_F(LQPTranslatorTest, PredicateNodeBetweenScan) {
   EXPECT_EQ(*table_scan_op->predicate(), *between_inclusive_(a, 42, 1337));
 }
 
+// Tests accessing the original LQP node after translation.
+TEST_F(LQPTranslatorTest, LqpNodeAccess) {
+  // clang-format off
+
+  // Predicates
+  {
+    auto predicate_node = PredicateNode::make(between_inclusive_(int_float_a, 42, 1337), int_float_node);
+    const auto op = LQPTranslator{}.translate_node(predicate_node);
+
+    const auto lqp_node = op->lqp_node;
+    const auto recovered_node = std::dynamic_pointer_cast<const PredicateNode>(lqp_node);
+    EXPECT_EQ(recovered_node, predicate_node);
+  }
+
+  // Joins
+  {
+    auto join_node = JoinNode::make(JoinMode::Inner, equals_(int_float_a, int_float2_a),
+                              int_float_node,
+                              int_float2_node);
+    const auto op = LQPTranslator{}.translate_node(join_node);
+
+    const auto lqp_node = op->lqp_node;
+    const auto recovered_node = std::dynamic_pointer_cast<const JoinNode>(lqp_node);
+    EXPECT_EQ(recovered_node, join_node);
+  }
+
+  // Joins
+  {
+    auto aggregate_node = AggregateNode::make(expression_vector(int_float_a, int_float_b),
+                                              expression_vector(sum_(int_float_a), sum_(int_float_b)),
+                                              int_float_node);
+    const auto op = LQPTranslator{}.translate_node(aggregate_node);
+
+    const auto lqp_node = op->lqp_node;
+    const auto recovered_node = std::dynamic_pointer_cast<const AggregateNode>(lqp_node);
+    EXPECT_EQ(recovered_node, aggregate_node);
+  }
+
+  // Validates
+  {
+    auto validate_node = ValidateNode::make(int_float_node);
+    const auto op = LQPTranslator{}.translate_node(validate_node);
+
+    const auto lqp_node = op->lqp_node;
+    const auto recovered_node = std::dynamic_pointer_cast<const ValidateNode>(lqp_node);
+    EXPECT_EQ(recovered_node, validate_node);
+  }
+
+  // clang-format off
+}
```

Comment:
Hm, I don't have a strong opinion on that, but maybe a test with more than one node instead of a (not complete) list of different operators?

Reply:
Will do.

Reply:
@Bensk1 : can you take another look?

Reply:
Please ignore the unnecessary brackets. Will be removed.